### ISSUE TITLE
UI rendering fail if $ref object in response extend object via "allOf"

### ISSUE THREAD

**Issue by gurubook** [swagger-api/swagger-ui#1677]
UI rendering fail if $ref object in response extend object via "allOf" 
The error reported is : "Uncaught TypeError: Cannot read property 'split' of null"

example to be added to swagger-js/test/spec/v2 
- singleAllOf.json

```
{
  "description": "A tag entity in the system",
  "allOf": [
      {
          "$ref": "single.json"
      },
      {
          "type": "object",
          "properties": {
            "lastName": {
              "type": "string"
            }
          }
      }
  ]
}
```
- spec4.json

```
{
  "swagger": "2.0",
  "info": {
    "description": "This is a sample server Petstore server.  You can find out more about Swagger at <a href=\"http://swagger.io\">http://swagger.io</a> or on irc.freenode.net, #swagger.  For this sample, you can use the api key \"special-key\" to test the authorization filters",
    "version": "1.0.0",
    "title": "Swagger Petstore",
    "termsOfService": "http://swagger.io/terms/",
    "contact": {
      "name": "apiteam@swagger.io"
    },
    "license": {
      "name": "Apache 2.0",
      "url": "http://www.apache.org/licenses/LICENSE-2.0.html"
    }
  },
  "host": "localhost:8000",
  "basePath": "/v2/api",
  "paths": {
    "/linkedAllOfResponse": {
      "get": {
        "parameters": [
          {
            "name": "status",
            "in": "query",
            "description": "Status values that need to be considered for filter",
            "required": false,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "$ref": "singleAllOf.json"
            }
          },
          "400": {
            "description": "Invalid status value"
          }
        }
      }
    }
  }
}
```

**Comment by maxatbrs** [comment]
Hey @gurubook I am having a similar issue. Here is a snippet with of my `swagger.json`

```
      "properties": {
        "name": {
          "type": "string"
        },
        "services": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/Nyv8jB879dx"
          }
        }
      },
```

When I browse to my Swagger endpoint in my app, I see this:

![screen shot 2016-01-21 at 3 49 02 pm](https://cloud.githubusercontent.com/assets/851978/12494139/9df76dae-c056-11e5-8878-c7a55fd81845.png)

I get the following error in my console:

    Uncaught TypeError: Cannot read property '$ref' of undefined

I am looking into this but if anyone has any suggestions, please let me know. Thanks!

**Comment by webron** [comment]
@maxatbrs - that doesn't look like the same issue. I'd suggest opening a separate ticket and provide the full spec so we can further investigate.

**Comment by maxatbrs** [comment]
Hey @webron absolutely. Is this the correct repo to place the issue in? Thanks!

**Comment by fehguy** [comment]
Hi @maxatbrs it's really hard for you to know this, so it's just fine that you opened it here.  The underlying issue will be in [swagger-js](https://github.com/swagger-api/swagger-js) so we'll open an issue there and close this one when it's merged.

**Comment by fehguy** [comment]
I've tested the previous spec error and it's working fine with master.  @maxatbrs please give more info, like your whole spec and I'll be able to figure it out.

**Comment by ralfhandl** [comment]
Seem to have the same or a similar error with swagger files using allOf and $ref (to mimic inheritance). Error showed up today after rebasing. Two examples attached:

[trippin.swagger.json.txt](https://github.com/swagger-api/swagger-ui/files/153664/trippin.swagger.json.txt)

[example.swagger.json.txt](https://github.com/swagger-api/swagger-ui/files/153661/example.swagger.json.txt)

Both files work fine in Swagger Editor, it just complains about unused definitions in allOf/1. 
Swagger UI loads TripPin file, shows "Finished Loading Resource Information. Rendering Swagger UI..." at top of page, and does not interpret GFM in descriptions.

Swagger UI gets stuck in Example file, shows "fetching resource list: http://localhost/examples/example.swagger.json; Please wait." and never shows actual UI.

Thanks in advance!
Ralf

**Comment by fehguy** [comment]
In `example.swagger.json.txt`, this construct:

```
        "Photo": {
          "description": "unfulfillable constraint, no inline representation of Edm.Stream",
          "type": "string",
          "enum": [
            null
          ]
        }
```

is invalid.  You cannot have `null` as an enum item, once removed it renders just fine.

**Comment by ralfhandl** [comment]
Hi Tony,

Thanks for the quick reply! After removing this construct from example.swagger.json.txt the browser tab now crashes when I try to load the file. New version of file is attached.
[example.swagger.json.txt](https://github.com/swagger-api/swagger-ui/files/154589/example.swagger.json.txt)

Also the TripPin.swagger.json.txt doesn't have this construct, and while it doesn't crash the browser tab, it is incompletely processed and GFM descriptions are shown with their "raw" value.

What else could be the source of the problems?

Thanks in advance!
Ralf

**Comment by fehguy** [comment]
Hi @ralfhandl, it seems to be working fine with master.  When you say it crashes out, what specifically is happening?

![image](https://cloud.githubusercontent.com/assets/249413/13654347/ac9278c8-e60c-11e5-84c3-ef76e892f804.png)

**Comment by ralfhandl** [comment]
What happens is that the banner shows, then apparently nothing happens for some time, then I see this:
![capture](https://cloud.githubusercontent.com/assets/951576/13663021/f7d096ac-e69e-11e5-9fd2-c01a48a3a0aa.PNG)

**Comment by fehguy** [comment]
I see, that's different from what I've been looking at.  I will try with json editor enabled.

**Comment by jonaswindey** [comment]
I've had this error and fixed it. It was because of an invalid config:

Not working:
```attributes: Joi.array().items(Joi.any()).description('Attributes'),```

Working:
```attributes: Joi.array().items(Joi.any().required()).description('Attributes'),```

The error here slipped in when upgrading Joi

**Comment by webron** [comment]
This should be resolved in 3.0. Please reopen otherwise.
